### PR TITLE
ref(ratelimits): Add a killswitch to optionmanager

### PR DESCRIPTION
**Pull request by AniketDas-Tekky**
`api.rate-limiter-activated` is a kill-switch for the global rate limiter. Disabling it will stop ALL rate limit enforcement, even if the endpoint flag `enforce_rate_limit` is True, the kill switch overrides it.
```
 killswitch | enforce_rate_limit | enforced?
--------------------------------------------
   True     |      False         |  False
   True     |       True         |   True
  False     |      False         |  False
  False     |       True         |  False
```

### CODE REVIEW

**Review by wedamija — APPROVED** [review]
lgtm, any need for a custom TTL for this option or is the default fine?

File: src/sentry/api/base.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ class Endpoint(APIView):
     # Default Rate Limit Values, override in subclass
     # Should be of format: { <http function>: { <category>: RateLimit(limit, window) } }
     rate_limits: Mapping[str, Mapping[RateLimitCategory | str, RateLimit]] = {}
-    enforce_rate_limit: bool = settings.SENTRY_RATELIMITER_ENABLED
+    enforce_rate_limit: bool = False
```
Comment (vladkluev) [inline src/sentry/api/base.py:106]: is this still used?

Comment (AniketDas-Tekky) [reply src/sentry/api/base.py:106]: Not right now, but I'm planning on deleting the `rate_limit_endpoint` decorator and using this instead as a sort of soft rollout.

**Comment by mgaeta** [comment]
Can I close this?

**Comment by AniketDas-Tekky** [comment]
> Can I close this?

Done.

**Pull request closed without merging**